base/visitor.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <sstream>
+#include <string>
+
+namespace base
+{
+class DebugPrintVisitor
```

Comment:
I propose to write a test on DebugPrintVisitor.

Reply:
Good idea, thanks!